include_core/OMR/Bytes.hpp

```diff
@@ -53,6 +53,25 @@ gibibytes(size_t n)
 	return n * mebibytes(1024);
 }
 
+#if defined(OMR_ENV_DATA64)
+static const size_t ALIGNMENT_MAX = size_t(1) << 63;
+static const size_t UNALIGNED_SIZE_MAX = size_t(1) << 63;
+#else
+static const size_t ALIGNMENT_MAX = size_t(1) << 31;
+static const size_t UNALIGNED_SIZE_MAX = size_t(1) << 31;
+#endif
+
+/// Round a size up to a multiple of alignment.
+/// Alignment must be a positive multiple of two, EG 1, 2, 4, 8, or 16.
+/// The maximum unaligned size is intrinsically related to the alignment.
+/// As a conservative measure, users should not use aligned sizes greater than
+/// UNALIGNED_MAX_SIZE, nor use alignments larger than ALIGNMENT_MAX.
+inline size_t
+align(size_t size, size_t alignment)
+{
+	return (size + (alignment - 1)) & ~(alignment - 1);
```

Comment:
Given the "shoot yourself in the foot" likelyhood of someone using this for non-powers of two, can we add an assert in here to check that `alignment` is a power of two rather than relying on the user reading the documentation. We can use something like [1] to check for it.

[1] https://graphics.stanford.edu/~seander/bithacks.html#DetermineIfPowerOf2

Reply:
+1.   That query alone should be a utility in this core library somewhere.  The power-of-2 question is already asked in the JIT, so we can eventually replace those questions with whatever becomes available in core.

Reply:
If this is going to be used by all components, including the GC, which currently has the exact code above I would rather keep the code to a minimum as to not impact the performance in tights loops in the GC that calls this code.  There is tons of complicated code that requires users to read the API documentation before they call it. If preferred we can rename the API to something that implies a power of 2 alignment but I would rather not slow down the code for no reason.

Reply:
Yeah, my intention was for this to be a "quick and dirty" call. Especially when you consider the use cases--fast path allocator for example--it makes sense to keep it small. 

I could split this into a safe `align` and an `alignUnsafe` helper. I can also add `isAligned(n, a)` and `isPow2(x)` to the header, while I'm here.

There aren't any assertion utilities in OMR core (yet, AFAIK), so any checks would have to be done with the standard C assert macro. cassert is disabled in release builds, which I think is the behaviour we want, although these kinds of assertions tend to mislead people when debugging.

Reply:
Do we want to accept an alignment of zero, or not?

Reply:
Yes do not accept 0